upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -1422,6 +1424,43 @@ func (n *nodeUpConfigBuilder) BuildConfig(ig *kops.InstanceGroup, apiserverAddit
 		}
 	}
 
+	if apiModel.UseJWTForBootstrap(cluster.Spec.GetCloudProvider()) {
+		if !ig.HasAPIServer() {
+			keyset := keysets["bootstrap-ca"]
+			if keyset == nil {
+				return nil, nil, fmt.Errorf("key bootstrap-ca not found")
+			}
+
+			if keyset.Primary != nil && keyset.Primary.PrivateKey != nil {
+				b, err := keyset.Primary.PrivateKey.AsBytes()
+				if err != nil {
+					return nil, nil, fmt.Errorf("unable to find privatekey: %w", err)
+				}
+
+				key, err := jwt.ParseRSAPrivateKeyFromPEM(b)
+				if err != nil {
+					return nil, nil, fmt.Errorf("parsing pem key failed: %w", err)
+				}
+				now := time.Now().UTC()
+				claims := jwt.MapClaims{
+					"ig":  ig.Name,
+					"iat": now.Unix(),
+					"nbf": now.Unix(),
+					"exp": now.Add(time.Hour).Unix(),
```

Comment:
the problem with this is that if someone/some reason will restart the instance. It cannot authenticate back:

```
I0118 07:52:14.295244       1 server.go:145] bootstrap 10.2.97.28:36072 verify err: validate: Token is expired
```

Why we are fetching the certificates again from kops-controller if node will restart? It do have certificates already in disk?